src/Datadog.Trace/DuckTyping/DuckType.Methods.cs

```diff
@@ -666,8 +666,68 @@ private static MethodInfo SelectTargetMethod(Type targetType, MethodInfo proxyMe
                     {
                         if (!candidateParamType.IsAssignableFrom(proxyParamType))
                         {
-                            skip = true;
-                            break;
+                            // Check if the parameter type contains generic types before skipping
+                            if (!candidateParamType.IsGenericType || candidateParamType.IsGenericType != proxyParamType.IsGenericType)
```

Comment:
The second condition will always have `candidateParamType.IsGenericType` being `true`, so I think this can be simplified:
```suggestion
                            if (!candidateParamType.IsGenericType || !proxyParamType.IsGenericType)
```